### PR TITLE
Nil plugin: Make purpose more clear in documentation.

### PR DESCRIPTION
**Pull request by kyrofa**
Some users were under the impression that nil could pull sources, when it can't. This PR resolves LP: [#1549691](https://bugs.launchpad.net/snapcraft/+bug/1549691) by hopefully making that more clear, and giving an example of what it CAN do.

### CODE REVIEW

**Comment by sergiusens** [comment]
By also failing on using keywords not available to the part we are golden here.

**Comment by kyrofa** [comment]
@sergiusens I thought that was something done in the new validator, but it doesn't seem so. Did I misunderstand that?

**Comment by kyrofa** [comment]
> I thought that was something done in the new validator, but it doesn't seem so. Did I misunderstand that?

It was, but not in the nil plugin, so fixed that here as well.

File: snapcraft/tests/test_pluginhandler.py [hunk 1]
```diff
@@ -1345,7 +1345,7 @@ def test_clean_stage(self):
             with self.subTest(key=key):
                 self.clear_common_directories()
 
-                schema = {'snap': {'type': 'array'}}
+                schema = {'stage': {'type': 'array'}}
```
Comment (sergiusens) [inline snapcraft/tests/test_pluginhandler.py]: what's this about?

Comment (kyrofa) [inline snapcraft/tests/test_pluginhandler.py]: A copy-paste error in a test that was only discovered as a result of the new `additionalProperties: false`.

**Comment by sergiusens** [comment]
Looks rather good; a test in would be good to capture the fact that source can't be used with nil; an integration test is fine if it is easier

**Comment by kyrofa** [comment]
> a test in would be good to capture the fact that source can't be used with nil; an integration test is fine if it is easier

Done. I did end up doing an integration test as the validation is handled in the pluginhandler but it depends on the nil plugin-- the test couldn't logically go in either place.

**Pull request merged**